spanner/google/cloud/spanner/database.py

```diff
@@ -79,6 +79,7 @@ def __init__(self, database_id, instance, ddl_statements=(), pool=None):
         self.database_id = database_id
         self._instance = instance
         self._ddl_statements = _check_ddl_statements(ddl_statements)
+        self._transaction_running = False
```

Comment:
@tseaver @jonparrott Are `Database` objects thread-safe? That would be surprising to me.

If they are, then Tres is probably correct that there really is not a whole lot we can do (and, considering the contortions required to get into this problem, that does not bother me much).

If they are not, then this solution will work.